### PR TITLE
Nicer error in the compatibility plugin with custom compilers

### PR DESCRIPTION
**Pull request by AbrilRBS**
Changelog: Bugfix: Nicer error in the compatibility plugin with custom compilers.
Docs: Omit

After seeing https://github.com/conan-io/conan/issues/13325, figured I'd improve the UX of the raised error. My first idea was to make the function return an empty list if the compiler was not known, but that would change current behaviour and so I decided to at least improve the generated exception, happy to discuss if going that way would work or not (Mostly so I get to learn about it a bit more :))

Also, I noticed that the plugin code itself is _not_ currently directly tested and we might want to look into improving that, but I tried and got hit a wall because of how the function is defined in the code, it makes it hard to test. We could get it to run by running the same code that the `test_migrations` code does, but I was not able to come up with a test that exercised the compatibility feature, so any help is welcomed!

### CODE REVIEW

**Comment by memsharded** [comment]
> After seeing https://github.com/conan-io/conan/issues/13325, figured I'd improve the UX of the raised error. My first idea was to make the function return an empty list if the compiler was not known, but that would change current behaviour and so I decided to at least improve the generated exception, happy to discuss if going that way would work or not (Mostly so I get to learn about it a bit more :))

Yes, definitely a better error until we discuss what would be a better default (taking into account that there will be users adding their own compilers to ``settings.yml``, that would hit this error).

> Also, I noticed that the plugin code itself is not currently directly tested and we might want to look into improving that, but I tried and got hit a wall because of how the function is defined in the code, it makes it hard to test. We could get it to run by running the same code that the test_migrations code does, but I was not able to come up with a test that exercised the compatibility feature, so any help is welcomed!

I think there are tests like ``TestCacheCompatibles`` that validate behavior of ``compatibility.py``, isn't that enough?

File: conans/client/graph/compatibility.py [hunk 1]
```diff
@@ -36,6 +37,8 @@ def cppstd_compat(conanfile):
         return []
     base = dict(conanfile.settings.values_list)
     cppstd_possible_values = supported_cppstd(conanfile)
+    if cppstd_possible_values is None:
```
Comment (memsharded) [inline conans/client/graph/compatibility.py:40]: Lets change it to return [] and warning

**Review by memsharded — CHANGES_REQUESTED** [review]

**Review by memsharded — APPROVED** [review]
I think a unit test would be great.

**Comment by AbrilRBS** [comment]
Done, I have added a case where it all works just to make sure

**Pull request merged**